zachsinterview/src/main/java/com/imprev/zachsinterview/thirdpartyapi/ThirdPartyDictionaryAPI.java

```diff
@@ -0,0 +1,54 @@
+package com.imprev.zachsinterview.thirdpartyapi;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * This is a third part library we are committed to using for internal storage, the relevant interface with some
+ * defined Exception types are available to understand how it is used.  This file should not be included in your review,
+ * it is only here for reference on how this API operates.
+ *
+ */
+public interface ThirdPartyDictionaryAPI {
```

Comment:
might want to check with the third party client if they have server/client API version, so we can check for API version compatibility before running any commands. It would make their lives easier as the update API version, and our lives easier so we can more easily detect when our calls are failing due to (our) obsolete client